Source/automap.cpp

```diff
@@ -382,18 +382,18 @@ WORD GetAutomapType(int x, int y, bool view)
 	WORD rv;
 
 	if (view && x == -1 && y >= 0 && y < DMAXY && automapview[0][y]) {
-		if (GetAutomapType(0, y, false) & (MAPFLAG_DIRT << 8)) {
+		if (GetAutomapType(0, y, false) & (MAPFLAG_DIRT * 256)) {

```

Comment:
Multiplication is not appropriate here and makes the code less clear. In this case we are referring to a 16-bit flag where the top 8 bits and lower 8 bits are different flags. This applies to most of the automap code and a few other places in this PR as well. There are many places where regular multiplication and division make sense though.

Reply:
We should use `std::bitset` for these in the (hopefully near) future.

Reply:
> We should use `std::bitset` for these in the (hopefully near) future.

I'm not sure what the performance will be like with bitset, we need to be careful with that.

What we can do for free is tag our flag enums with the appropriate annotation for compilers that support it

https://clang.llvm.org/docs/AttributeReference.html#flag-enum